server/integration/event-logger/src/main/java/org/infinispan/server/eventlogger/ServerEventLogger.java

```diff
@@ -95,7 +95,7 @@ TimeService getTimeService() {
 
    @Override
    public List<EventLog> getEvents(Instant start, int count, Optional<EventLogCategory> category, Optional<EventLogLevel> level) {
-      List<EventLog> events = new ArrayList<>();
+      List<EventLog> events = Collections.synchronizedList(new ArrayList<>());
       AtomicReference<Throwable> throwable = new AtomicReference<>();
       try {
          cacheManager.executor().submitConsumer(m -> {
```

Comment:
The throwable is wrapped in a CacheException and thrown. The outer try-catch(Exception) will wrap it again. I think we should have a separate catch for CacheException and Exception and avoid the double wrapping.